ouroboros_macro/src/generate/summon_checker.rs

```diff
@@ -33,9 +33,8 @@ pub fn generate_borrowchk_summoner(info: &StructInfo) -> Result<TokenStream, Err
             }
             code.push(quote! { let #field_name = #builder_name (#(#builder_args),*); });
         }
-        if field.is_borrowed() {
-            let boxed = field.boxed();
-            code.push(quote! { let mut #field_name = #boxed; });
+        if field.is_mutably_borrowed() {
+            code.push(quote! { let mut #field_name = #field_name; });
         };
         if !field.is_mutably_borrowed() {
```

Comment:
Actually, I wouldn’t complain about an additional `let #field_name = #field_name` in this “`!field.is_mutably_borrowed()`” case, because it would ensures that you’re getting _exactly_ the same drop order of local variables as the resulting struct will be having. Not that I’d expect it to ever make an actual difference.

Also this could become an `else` of the previous `if`.